Emby.Server.Implementations/Library/IgnorePatterns.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Linq;
+using DotNet.Globbing;
+
+namespace Emby.Server.Implementations.Library
+{
+    public static class IgnorePatterns
+    {
+        /// Files matching these glob patterns will be ignored
+        public static readonly string[] Patterns = new string[]
+        {
+            "**/small.jpg",
+            "**/albumart.jpg",
+            "**/*sample*",
+
+            // Directories
+            "**/metadata/**",
+            "**/ps3_update/**",
+            "**/ps3_vprm/**",
+            "**/extrafanart/**",
+            "**/extrathumbs/**",
+            "**/.actors/**",
+            "**/.wd_tv/**",
+
+            // WMC temp recording directories that will constantly be written to
+            "**/TempRec/**",
+            "**/TempSBE/**",
+
+            // Synology
+            "**/eaDir/**",
+            "**/@eaDir/**",
+            "**/#recycle/**",
+
+            // Qnap
+            "**/@Recycle/**",
+            "**/.@__thumb/**",
+            "**/$RECYCLE.BIN/**",
+            "**/System Volume Information/**",
+            "**/.grab/**",
+
+            // MacOS
+            "**/._*", // https://apple.stackexchange.com/questions/14980/why-are-dot-underscore-files-created-and-how-can-i-avoid-them
+
+            // thumbs.db
+            "**/thumbs.db",
+
+            // bts sync files
+            "**/*.bts",
+            "**/*.sync",
+        };
+
+        private static readonly Glob[] _globs;
+
+        static IgnorePatterns()
```

Comment:
static Constructors shouldn't be used, though this is much cleaner then the alternative.

alternate method:
```c#
private static readonly GlobOptions _globOptions = new GlobOptions
{
	Evaluation =
	{
		CaseInsensitive = true
	}
};

private static readonly Glob[] _globs = Patterns.Select(p => Glob.Parse(p, _globOptions)).ToArray();
```

Reply:
So which should I use?

Reply:
Leaving it up to the rest of @jellyfin/backend

Reply:
I have a sort of hatred against static constructors due to their sort of implicit behavior. They have cost me a great deal of hair when figuring out why stuff was not working. Now in this case it won't break right now I don't think but nice is different.

Reply:
Eventually these patterns should be loaded from the config. Potentially even supporting something like a `.jfignore` file. So ideally a instance of this should be dependency injected.